api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java

```diff
@@ -0,0 +1,190 @@
+package org.openmrs.module.locationbasedaccess.aop;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.openmrs.Location;
+import org.openmrs.Person;
+import org.openmrs.PersonAttribute;
+import org.openmrs.PersonAttributeType;
+import org.openmrs.User;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.PersonService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.LocationBasedAccessConstants;
+import org.openmrs.module.locationbasedaccess.utils.LocationUtils;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class LocationUtilsTest extends BaseModuleContextSensitiveTest {
+
+	private PersonService personService;
+	private LocationService locationService;
+	private UserService userService;
+	private static final String XML_FILENAME_WITH_PERSON_DATA = "include/LocationUtilTestData.xml";
+	private static final int DEMO_PERSON3_ID = 3;
+	private static final int DEMO_PERSON2_ID = 2;
+	private static final int DEMO_PERSON5_ID = 6005;
+	private static final int DEMO_PERSON6_ID = 6006;
+	private static final int DEMO_LOCATION1_ID = 1;
+	private static final int DEMO_LOCATION2_ID = 2;
+	private static final int DEMO_USER1_ID = 6001;
+	private static final int DEMO_PERSONATTRIBUTE1_ID = 10;
+
+	public LocationUtilsTest(){
+
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		userService = Context.getUserService();
+		personService = Context.getPersonService();
+		locationService = Context.getLocationService();
+		executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+	}
+
+	@Test
+	public void getPersonLocation_shouldReturnPersonLocation(){
+		Person person3 = personService.getPerson(DEMO_PERSON3_ID);
+		Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+		PersonAttributeType personAttributeType = personService.getPersonAttributeType(DEMO_PERSONATTRIBUTE1_ID);
+		PersonAttribute personAttribute = new PersonAttribute(personAttributeType,location1.getUuid());
+		person3.addAttribute(personAttribute);
+		Location location3 = LocationUtils.getPersonLocation(person3);
+		assertEquals(location3,location1);
```

Comment:
Did you confirm that the person doesn't have this person Attribute or Location before you adding the person attribute?

Reply:
Added `AssertNull` for checking attributes. Should be fine now.